src/test/run-make-fulldeps/instrument-coverage/Makefile

```diff
@@ -0,0 +1,67 @@
+# needs-profiler-support
+
+-include ../tools.mk
+
+# This test makes sure that LLVM coverage maps are genereated in LLVM IR.
+
+COMMON_FLAGS=-Zinstrument-coverage
+
+# LLVM doesn't yet support instrumenting binaries that use unwinding on MSVC:
+# https://github.com/rust-lang/rust/issues/61002
+#
+# Things work fine with -Cpanic=abort though.
+ifdef IS_MSVC
+WINDOWS_NOT_YET_SUPPORTED=true
+endif
+
+all:
+  # FIXME(richkadel): Debug the following problem, and reenable on Windows.
+  # Abort without failing if testing on Windows. The current implementation
+	# generates a segfault when running the instrumented `main` executable,
+	# after the `main` program code executes, but before the process terminates.
+	# This most likely points to a problem generating the LLVM "main.profraw"
+	# file.
+	if [[ "$(WINDOWS_NOT_YET_SUPPORTED)" == true ]]; then \
+	  >&2 echo 'Skip test: -Zinstrument-coverage not yet available on Windows'; \
+	  exit 0; \
+	fi
+
+	# Compile the test program with instrumentation, and also generate LLVM IR
+	$(RUSTC) $(COMMON_FLAGS) main.rs
+
+	# Run it in order to generate some profiling data,
+	# with `LLVM_PROFILE_FILE=<profdata_file>` environment variable set to
+	# output the coverage stats for this run.
+	LLVM_PROFILE_FILE="$(TMPDIR)"/main.profraw \
+	  $(call RUN,main)
+
+	# Postprocess the profiling data so it can be used by the llvm-cov tool
+	"$(LLVM_BIN_DIR)"/llvm-profdata merge --sparse \
+	  "$(TMPDIR)"/main.profraw \
+		-o "$(TMPDIR)"/main.profdata
+
+	# Generate a coverage report using `llvm-cov show`. The output ordering
+	# can be non-deterministic, so ignore the return status. If the test fails
+	# when comparing the JSON `export`, the `show` output may be useful when
+	# debugging.
+	"$(LLVM_BIN_DIR)"/llvm-cov show \
+	  --Xdemangler="$(RUST_DEMANGLER)" \
+	  --show-line-counts-or-regions \
+	  --instr-profile="$(TMPDIR)"/main.profdata \
+		"$(TMPDIR)"/main \
```

Comment:
```suggestion
		$(call BIN,"$(TMPDIR)"/main) \
```

Reply:
fixed

Reply:
I looked at both #61002 and the referenced LLVM bug report, and still can't reconcile this with the failure location indicated in Tyler's screenshot. I certainly don't want to rule it out, but I rather than try to investigate this, and possible workarounds, via `bors try`, let's merge this PR as-is with Windows temporarily disabled for the instrument-coverage test, and then debug this on local MSVC-equipped machines. (My Windows VM is almost ready.)

Reply:
> > FWIW `actual_export_coverage.json` has CRLF EOL characters.
> 
> Should now be resolved by adding -Z option to the diff

Nope! Apparently MacOS diff doesn't recognize -Z. It looks like `diff --strip-trailing-cr` is a common solution to this. I'm making that change.

@mati865 @tmandry  -  It won't matter for this PR since we're skipping the test on Windows, but do you know if `diff --strip-trailing-cr` flag is available on MSVC?